SMS_D_Ln9_Mmpi-serial.f19_g16_rx1.A fails to build with scorpio classic

Scorpio classic fails to build with the following test case that uses the mpi-serial library.

```
/sandbox/jayesh/ACME_test5/externals/scorpio_classic/pio/pio_types.F90:371:48:

    integer, public, parameter :: PIO_FILL_CHAR = nf90_fill_char
                                                1
Error: Can't convert CHARACTER(1) to INTEGER(4) at (1)
make[2]: *** [pio/CMakeFiles/pio.dir/pio_types.F90.o] Error 1
make[1]: *** [pio/CMakeFiles/pio.dir/all] Error 2

```
The issue can be recreated by creating a simple X case with the mpi-serial library,
```
./create_newcase -case test_X_f19_g16 -compset X -res f19_g16 -mpilib mpi-serial
```

The issue is caused by the different types used by NetCDF libraries for character fillvalues,

* The Fortran 77 NetCDF library (netcdf.inc) defines nf_fill_char as an integer
* The Fortran 90 NetCDF library (netcdf.mod) defines nf90_fill_char as a character

The build fails when using the Fortran 90 interface (SCORPIO defines PIO_FILL_CHAR as an integer and Fortran 90 NetCDF library defines nf90_fill_char as a character - type mismatch when assigning one to the other). The NetCDF libraries on anlworkstation built with the mpi-serial library have the F90 interfaces available and hence the issue shows up with the mpi-serial library.

Also note that Scorpio does not use the NetCDF Fortran 90 modules (always uses netcdf.inc) and hence does not have the issue

The character fillvalues were introduced in PR #176

--- a/pio/pio_types.F90
+++ b/pio/pio_types.F90
@@ -368,7 +368,7 @@ module pio_types
    integer, public, parameter :: PIO_MAX_VAR_DIMS = nf90_max_var_dims
    integer, public, parameter :: PIO_64BIT_OFFSET = nf90_64bit_offset
    integer, public, parameter :: PIO_64BIT_DATA = 0
-   integer, public, parameter :: PIO_FILL_CHAR = nf90_fill_char
+   integer, public, parameter :: PIO_FILL_CHAR = iachar(nf90_fill_char)
    integer, public, parameter :: PIO_FILL_INT = nf90_fill_int
    real, public, parameter :: PIO_FILL_FLOAT = nf90_fill_float
    double precision, public, parameter :: PIO_FILL_DOUBLE = nf90_fill_double
@@ -387,7 +387,7 @@ module pio_types
    integer, public, parameter :: PIO_NOWRITE = 21
    integer, public, parameter :: PIO_64BIT_OFFSET = 0
    integer, public, parameter :: PIO_64BIT_DATA = 0
-   integer, public, parameter :: PIO_FILL_CHAR = achar(0)
+   integer, public, parameter :: PIO_FILL_CHAR = 0
    integer, public, parameter :: PIO_FILL_INT = -2147483647
    real, public, parameter :: PIO_FILL_FLOAT = 9.9692099683868690e+36
    double precision, public, parameter :: PIO_FILL_DOUBLE = 9.9692099683868690e+36

